plugins/sysinfo/sysinfo.cpp

```diff
@@ -338,7 +342,9 @@ getWmiInfo (int mode)
 				hr = pclsObj->Get (L"Name", 0, &vtProp, 0, 0);
 				break;
 		}
-		WideCharToMultiByte (CP_ACP, 0, vtProp.bstrVal, -1, buffer, SysStringLen (vtProp.bstrVal)+1, NULL, NULL);
+		buffer_iter += WideCharToMultiByte(CP_ACP, 0, vtProp.bstrVal, -1, buffer_iter, WMI_INFO_BUFFER_LENGTH - (buffer_iter - buffer + 2), NULL, NULL) - 1;
```

Comment:
Alright, I resolved the buffer overflow and added made buffer null-initialized to avoid any string function issues.